package/piksi_system_daemon/src/main.c

```diff
@@ -191,6 +211,37 @@ int main(void)
   SETTING_NOTIFY("ethernet", "netmask", eth_netmask, TYPE_STRING, eth_ip_config_notify);
   SETTING_NOTIFY("ethernet", "gateway", eth_gateway, TYPE_STRING, eth_ip_config_notify);
 
+  char name_string[64];
+  if (file_read_string("/img_tbl/boot/name",
+                       name_string, sizeof(name_string)) == 0) {
+    /* sw_build_id contains the full image name */
+    static char sw_build_id[sizeof(name_string)];
+    strncpy(sw_build_id, name_string, sizeof(sw_build_id));
+    READ_ONLY_PARAMETER("system_info", "sw_build_id", sw_build_id, TYPE_STRING);
+
+    /* sw_version contains everything before the git hash */
+    static char sw_version[sizeof(name_string)];
+    strncpy(sw_version, name_string, sizeof(sw_version));
+    char *sep = strstr(sw_version, "-g");
+    if (sep != NULL) {
+      *sep = 0;
+    }
+    READ_ONLY_PARAMETER("system_info", "sw_version", sw_version, TYPE_STRING);
+  }
+
+  char timestamp_string[32];
+  if (file_read_string("/img_tbl/boot/timestamp",
+                       timestamp_string, sizeof(timestamp_string)) == 0) {
+    static char sw_build_date[128];
+    time_t timestamp = strtoul(timestamp_string, NULL, 10);
+    if (strftime(sw_build_date, sizeof(sw_build_date),
+                 "%Y-%m-%d %H:%M:%S %Z", localtime(&timestamp)) == 0) {
+      printf("error parsing timestamp\n");
+    } else {
+      READ_ONLY_PARAMETER("system_info", "sw_build_date", sw_build_date, TYPE_STRING);
+    }
```

Comment:
Lastly, can we add SBP log_infos with both of these settings when the device boots up?  In this way every version string  (nap, gnss, and buildroot) is printed on bootup.